world/handlers.go

```diff
@@ -176,6 +186,38 @@ func (server *Server) handlePartyEvent(conn mnet.Server, reader mpacket.Reader)
 		p.WriteBytes(reader.GetBuffer())
 		server.channelBroadcast(p)
 	default:
-		log.Println("Unkown party event type:", op)
+		log.Println("Unknown party event type:", op)
 	}
 }
+
+func (server *Server) handleChangeRate(conn mnet.Server, reader mpacket.Reader) {
+	mode := reader.ReadByte()
+	rate := reader.ReadFloat32()
+
+	switch mode {
+	case 1:
+		server.Info.Rates.Exp = rate
+	case 2:
+		server.Info.Rates.Drop = rate
+	case 3:
+		server.Info.Rates.Mesos = rate
+	}
+
+	h := sha512.New()
+	h.Write([]byte(fmt.Sprintf("exp %.2f drop %.2f mesos %.2f", server.Info.Rates.Exp, server.Info.Rates.Drop, server.Info.Rates.Mesos)))
+	hRates := hex.EncodeToString(h.Sum(nil))
+
+	if hRates != server.Info.DefaultRatesChecksum { // Rates event
```

Comment:
Why are the values being hashed instead of comparing via an if statement? The hash will be slower. Something like:
```
if server.Info.Rates.DefaultExp != server.Info.Rates.Exp || 
    server.Info.Rates.DefaultMesos != server.Info.Rates.Mesos || ... {
}
``

Reply:
it seemed like a nasty if statement that i wanted to avoid, but i guess hashing is way more computionally expensive